### DISCUSSION TITLE
[1.1.0-pre3][WindowsPackageType none] How to avoid closing the app?

### DISCUSSION THREAD

**Discussion by sude22** [microsoft/WindowsAppSDK#2480] [Q&A] [answered]
Hi,
I have a WinUI3 nonpackaged app.
How could I prevent the app from closing if the user clicks the X?
In UWP you could register to
SystemNavigationManagerPreview.GetForCurrentView().CloseRequested +=
but thats not working in WinUI3 nonpackaged app.

**Comment by castorix** [comment, marked as answer]
You can use [AppWindow.Closing Event](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.windowing.appwindow.closing?view=windows-app-sdk-1.0)
(_args.Cancel = true;_ to prevent closing)

**Comment by sude22** [comment]
@castorix 
Thank you, that works.
